CT-1870 Remove requestor name from email personalisation

stephenrichards opened the pull request.

aliuk2012 commented on app/policies/team_policy.rb:
```diff
@@ -4,36 +4,64 @@ class TeamPolicy < ApplicationPolicy
 
   def index?
     clear_failed_checks
-    check_user_is_a_manager
```
If theres no checks is it worth even having this policy?

aliuk2012 commented on app/controllers/teams_controller.rb:
```diff
@@ -3,8 +3,8 @@ class TeamsController < ApplicationController
   def index
-    @teams = BusinessGroup.all
-    authorize @teams.first
+    authorize Team
```
Do we need to check user is authorised to view the index page? I would think that we are more concerned about the list of teams in the list and whether they can view those teams

misaka commented on spec/controllers/teams_controller_spec.rb:
```diff
@@ -2,58 +2,154 @@
+      end
 
-      before { sign_in manager }
+      it 'renders the index template' do
```
As we discussed in person, I feel a bit apprehensive about shoving all these specs into this controller spec after our experience with `cases_controller_spec.rb`, which currently stands at 1434 lines long, and that's with 3 of the verbs broken out into separate files. Can I propose that we:

1. change the organisation of the contexts so that instead of describing the verbs within the context of the user, we have the context of the user within the description of the verb. i.e. instead of this:
```
context 'manager' do
  describe 'GET show' do ...
  describe 'POST create' do ...
end

context 'responder' do
  describe 'GET show' do ...
  describe 'POST create' do ...
end
```
we do this:
```
describe 'GET show' do
  context 'manager' do ...
  context 'responder' do ...
end

describe 'POST create' do
  context 'manager' do ...
  context 'responder' do ...
end
```
I understand that this means that we won't be able to re-use quite as much setup, but we can mitigate that by using more fixtures (with `let` statements), or even better, my next point:

2. Move larger verbs, or maybe even every verb, out into a separate file. For examples have a look at the specs in `spec/controllers/cases_controller/*`

stephenrichards committed:
debugging statemnets

stephenrichards committed:
CT-1870 REmove requestor name from email personalisation

ICO cases do not have a name field, which was causing notification
emails for ICO cases to silently fail.  It was decided to remove
name from all notifications - the mails contain a link to the case
where the reciipient can see all days in any case.

stephenrichards committed:
Merge branch 'master' into debugging-mailer

stephenrichards committed:
Merge branch 'master' into debugging-mailer

aliuk2012 committed:
Merge branch 'master' into debugging-mailer

aliuk2012 committed:
Merge branch 'master' into debugging-mailer
